### DISCUSSION TITLE
Find the edge direction

### DISCUSSION THREAD

**Discussion by MaxDragonheart** [networkx/networkx#6470] [Q&A]
I'm using `all_shortest_paths` on  a `MultiDiGraph`. Is there a way to know which edge direction is used to define the shortest path?

**Comment by dschult** [comment]
The order of the nodes gives the direction of the path.  For example the path `[4, 6, 3, 4, 2]` corresponds to the edges `[(4, 6), (6, 3), (3, 4), (4, 2)]`. You can get the edges from the path using `path_edges = nx.utils.pairwise(path)'

**Reply by MaxDragonheart** [reply]
Thank you @dschult . My question born because I need to know if there is a preferred direction on my network. My network is a real urban network and I need to find how many times is used every single edge and in which direction(e.g. North -> South, West -> East). I'm be able to find how many times is used every edges but I can't find a solution for the direction.
Do you have any suggestion?

**Reply by dschult** [reply]
What do you mean by "how many times?"  That depends on the traffic levels. right?

**Reply by MaxDragonheart** [reply]
No, it doesn't depends on traffic levels but only on how many paths use the single edge. I start the shortest path analysis from a GeoDataFrame from which I make a `MultiDiGraph`. One of the attributes that I add to an edge is the ID of every roads; others are length, time travel and the direction: 1 for one-way and 0 for two-ways. So, at the end, when I have the shortest path, I extract the edges and with a join I can know which ID is used. 

I know which road(edge) is used and how many times, but I don't know in which direction. Yes I know that the road ID1(e.g.) is bidirectional, but which of his directions is used more that the other? This is my question and my problem.

Hope I'm more clear now, my english is not perfect...
